src/main.go

```diff
@@ -56,6 +56,24 @@ func getCname(path string) (string, string, error) {
 	return cname, version, nil
 }
 
+func checkEFI(expected_loader_entry string) error {
+	_, err := os.Stat("sys/firmware/efi")
+	if err != nil {
+		return errors.New("not EFI booted")
+	}
+
+	data, err := os.ReadFile("/sys/firmware/efi/efivars/LoaderEntrySelected-4a67b082-0a4c-41cf-b6c7-440b29bb8c4f")
```

Comment:
I'm sure that uuid is there for a good reason, but that reason is not obvious to me. is this a 'real' constant?

Reply:
yep, in UEFI all variables are stored under a vendor id and a key, Linux exposes the vars as key-vendor_id. that's the id systemd-boot has picked for its varibles